model/task_lifecycle.go

```diff
@@ -49,9 +49,19 @@ func SetActiveState(t *task.Task, caller string, active bool) error {
 			if err != nil {
 				return errors.Wrapf(err, "error getting tasks '%s' depends on", t.Id)
 			}
-			tasksToActivate = append(tasksToActivate, deps...)
+			for _, dep := range deps {
```

Comment:
should we be checking if it's part of a single host task group here?

Reply:
We do check at line 55 if it's in the same task group as t which I was  thinking implied that it was in a single host task group but I can actually make that more explicit